docs/guides/self_hosting.md

```diff
@@ -200,4 +200,7 @@ $ curl -X POST 'http://localhost:5000/v1/llm/completions-sync?model_endpoint_nam
 You should get a response similar to:
 ```
 {"status":"SUCCESS","outputs":[{"text":". Tell me a joke about AI. Tell me a joke about AI. Tell me a joke about AI. Tell me","num_completion_tokens":30}],"traceback":null}
-```
\ No newline at end of file
+```
+
+### Pointing LLM Engine client to use self-hosted infrastructure
+The `llmengine` client makes requests to Scale AI's hosted infrastructure by default. You can have `llmengine` client make requests to your own self-hosted infrastructure by setting the `LLM_ENGINE_BASE_PATH` environment variable to the URL of the `llm-engine` pod. The exact URL of `llm-engine` pod depends on your Kubernetes cluster networking setup.
```

Comment:
I'm not 100% sure if this would just work out of the box as Spellbook API might be different from the API exposed by llm-engine gateway?

Reply:
Does having people use https://github.com/scaleapi/launch-python-client and point the `gateway_endpoint` to self-hosted llm-engine be better?

Reply:
I'd have thought that we'd have wanted to have `llmengine` client be able to point to `llmengine` server directly? in the sense that if this doesn't work OOB we should make it work OOB?

IMO having people use `launch-python-client` seems kinda ugly

Reply:
Let's not mention `launch-python-client` here. FWIW, the EGP APIs do have a `/llm`-prefixed set of APIs that are at parity with the LLM Engine APIs, so having the LLM Engine client point to either the self-hosted or EGP-hosted one should still work. There is a _separate_ set of EGP-specific Completion APIs, which is not at play here. cc @felixs8696

Reply:
Ok! Then I'll keep the documentation as it is and only use `llmengine` client

Reply:
> Spellbook API might be different from the API exposed by llm-engine gateway

we intentionally keep them the same.